File: signal-processor/trigger_demo.js
const http = require('http');
const https = require('https');

const crypto = require('crypto');

// Change this to your production backend URL if testing production
const BACKEND_URL = process.env.BACKEND_URL || 'http://localhost:3001';

const data = JSON.stringify({
    "event_id": crypto.randomUUID(),
    "node_name": "Bengaluru_Alpha",
    "city": "Bengaluru",
    "timestamp": new Date().toISOString(),
    "peak_frequency_mhz": 14.235,
    "peak_power_db": -55.4,
    "bandwidth_hz": 3500,
    "duration_ms": 120,
    "threat_level": "high",
    "signal_class": "Unidentified Burst",
    "is_burst": true,
    "is_licensed": false,
    "is_frequency_hopping": false,
    "classifier_confidence": 92.5,
    "fingerprint_hash": "a1b2c3d4e5f6a1b2c3d4e5f6a1b2c3d4e5f6a1b2c3d4e5f6a1b2c3d4e5f6a1b2",
    "psd_array": [
        25.4, 28.1, 35.6, 50.2, 85.9, 95.4, 88.2, 55.1, 30.5, 26.2, 24.1
    ]
});

console.log("Preparing to send highly anomalous threat vector to the ingestion engine...");

const url = new URL(`${BACKEND_URL}/api/anomalies/ingest`);
const client = url.protocol === 'https:' ? https : http;

const options = {
    hostname: url.hostname,
    port: url.port,
    path: url.pathname,
    method: 'POST',
    headers: {
        'Content-Type': 'application/json',
        'Content-Length': data.length
    }
};

const req = client.request(options, (res) => {
    let responseBody = '';
    
    res.on('data', (chunk) => {
        responseBody += chunk;
    });

    res.on('end', () => {
        if (res.statusCode >= 200 && res.statusCode < 300) {
            console.log("✅ CRITICAL THREAT SUCCESSFULLY INJECTED!");
            console.log("Look at the dashboard NOW to see the detection.");
        } else {
            console.log(`❌ Failed to inject threat: ${res.statusCode}`);
            console.log(responseBody);
        }
    });
});

req.on('error', (error) => {
    console.error(`Error connecting to backend: ${error.message}`);
});

req.write(data);
req.end();

File: signal-processor/trigger_demo.py
import requests
import os
import uuid
from datetime import datetime, timezone
from dotenv import load_dotenv

load_dotenv()
BACKEND_URL = os.getenv("BACKEND_URL", "http://localhost:3001")

DEMO_FINGERPRINT_HASH = (
    "a1b2c3d4e5f6a1b2c3d4e5f6a1b2c3d4e5f6"
    "a1b2c3d4e5f6a1b2c3d4e5f6a1b2c3d4e5f6a1b2"
)

payload = {
    "event_id": str(uuid.uuid4()),
    "node_name": "Delhi-Alpha",
    "city": "Delhi",
    "timestamp": datetime.now(timezone.utc).isoformat(),
    "peak_frequency_mhz": 14.235,
    "peak_power_db": -67.6,
    "bandwidth_hz": 2400.0,
    "duration_ms": 298.0,
    "psd_array": [50.0] * 256,
    "fingerprint_hash": DEMO_FINGERPRINT_HASH,
    "signal_class": "BURST_TRANSMISSION",
    "is_licensed": False,
    "is_burst": True,
    "is_frequency_hopping": False,
    "classifier_confidence": 0.68,
    "threat_level": "medium",
    "assessment": {
        "threat_level": "medium",
        "pattern_analysis": (
            "Seventh confirmed occurrence of burst "
            "on 14.235 MHz. 47-hour cycle now highly "
            "consistent across all detections."
        ),
        "recommended_action": (
            "Flag frequency for continued monitoring. "
            "Pattern suggests scheduled transmission."
        ),
        "confidence_adjustment": 3.1,
        "model_used": "llama3-70b-8192",
        "provider": "groq",
        "latency_ms": 847,
        "used_fallback": False
    }
}

resp = requests.post(
    f"{BACKEND_URL}/api/anomalies/ingest",
    json=payload,
    timeout=10
)

if resp.status_code == 201:
    print("Demo detection sent successfully.")
    print("Watch the Fingerprint Vault — "
          "Delhi 14.235 MHz confidence should jump.")
else:
    print(f"Failed: {resp.status_code} {resp.text}")
